src/chart/generateCategoricalChart.js

```diff
@@ -44,6 +44,7 @@ const generateCategoricalChart = (ChartComponent, GraphicalChild) => {
     static propTypes = {
       ...ChartComponent.propTypes,
       syncId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+      compact: PropTypes.bool,
```

Comment:
> All those exclusions are specifically made for the "panoramic brush" use case. Do you think this is way too specific? When looking back into this, I found myself may still need grid inside the panorama. Maybe it's better to extend this prop to an array (of chart parts to be excluded, like ['grid', 'chart', 'children']... a little tricky...)?
>
> Anyway for this MR, this prop is necessary, since when cloning element in Brush: https://github.com/recharts/recharts/pull/586/files/6c29a98d8e445ffe456d968cd9b0a3080e47b5bb#diff-44d40453cfbdcbd566bcc8d82acd5c5bR306 , I don't have any chance to exclude parts inside the wrapping Surface.

As what I've done in e597e3b, `compact=true` now would render complete svg contents within the wrapping `<Surface />`. 

If a user would like to render a panoramic brush, he needs to specify a certain Chart as child of `<Brush />`. Brush will ensure that to `compact=true` is set to its containing Chart, but user needs to specify what the contents are inside the Chart:

```jsx
<Brush>
  <AreaChart>
    <CartesianGrid />
    <YAxis hide domain={['auto', 'auto']} />
    <Area dataKey="price" stroke="#ff7300" fill="#ff7300" dot={false} />
  </AreaChart>
</Brush>
```

Please refer to a complete demo in 7470e8e.

Anyway, Tooltip, Legend and event handlers will be ignored when `compact=true` is set to the Chart.